src/checkers/rustchecker.py

```diff
@@ -0,0 +1,37 @@
+import toml
+import logging
+import urllib.request
+
+from src.lib import utils
+from src.lib.externaldata import Checker
+
+log = logging.getLogger(__name__)
+
+
+class RustChecker(Checker):
+    @staticmethod
+    def _should_check(external_data):
+        return external_data.checker_data.get("type") == "rust"
+
+    def check(self, external_data):
+        if not self._should_check(external_data):
+            log.debug("%s is not a rust type ext data", external_data.filename)
+            return
+
+        channel = external_data.checker_data["channel"]
+        package_name = external_data.checker_data["package"]
+        target_name = external_data.checker_data["target"]
+
+        url = f"https://static.rust-lang.org/dist/channel-rust-{channel}.toml"
+
+        log.debug("Getting extra data info from %s; may take a while", url)
+        with urllib.request.urlopen(url) as response:
+            data = toml.loads(response.read().decode())
+
+        package = data["pkg"][package_name]
+        target = package["target"][target_name]
+        if target["available"]:
+            new_version = utils.get_extra_data_info_from_head(target["xz_url"])
```

Comment:
Is there any to download the archive and not just use provided `xz_url` and `xz_hash` as is?

Reply:
`xz_hash` is actually used, see the line below this one

I have done it like this because it is the same way here:
https://github.com/flathub/flatpak-external-data-checker/blob/69ea8b5790853c0069f0a5ed57f13fb62fddb3a6/src/checkers/firefoxchecker.py#L167-L168

I think it's there to get the size of the the file, not sure if that’s optional here.
But in any case the head request shouldn't download the file

Reply:
Size is useless for archive type sources anyway. I guess we shouldn't add it for anything but `extra-data`, but that's a separate issue.

Reply:
I am not sure how to construct a ExternalFile without a size.
Should I use a some fixed bogus value to avoid the inexpensive `HEAD` request?

Reply:
Maybe not. We could rewrite this later once we get rid of spurious `size` property.

Reply:
Now you can construct `ExternalFile`  manually and set `None` as `size` value (after rebasing onto master).

Reply:
done